carbon-design-system/carbon-addons-iot-react
# fix(tablecard): in dashboard editor

Closes #

**Summary**

- There were two problems in using the TableCards in the DashboardEditor
- The first was that in certain instances a pageSize of 0 was being passed, which causes the Pagination component from carbon to cycle indefinitely.  Also there was a small bug where the whole table rows wouldn't render and there was whitespace underneath
- The second was that if the columns in the table changed, it would render duplicate columns due to a bug with the underlying Table where if you change the order and number of columns it renders duplicates

**Change List (commits, features, bugs, etc)**

- fix(TableCard): use key to throw away the table and rerender if the pageSize or the number of columns change.
- fix(TableCard): we can fit two more rows in the editable card, fix the numberOfRowsPerPage calculation
- fix(timeSeriesUtils): increase the generated number of rows to 100 so we can handle all the possible sizes of a table card

**Acceptance Test (how to verify the PR)**

- Verify the old TableCard stories
- Verify that if you add a TableCard through the DashboardEditor, and change the columns, it renders successfully with the correct columns. Make sure if you change the size of the TableCard, the new number of rows per page renders
